### ISSUE TITLE
Missing of headers support on UMF messages affecting hydra-router response on legacy environments

### ISSUE THREAD

**Issue by jkyberneees** [flywheelsports/hydra-router#87]
Hydra stands on the UMF message spec for req/res as well as for real-time messaging. UMF messages does not consider possible HTTP headers except 'Content-Type' and 'Authorization'. On legacy environments, to integrate the hydra can suppose a problem because HTTP headers are heavily used. 

As an example, let's analyze the response headers of an app built with hapi.js:
> http://localhost:3000/echo/hello
```
Connection →keep-alive
Date →Wed, 19 Apr 2017 08:26:40 GMT
Transfer-Encoding →chunked
cache-control →no-cache
content-encoding →gzip
content-type →application/json; charset=utf-8
vary →accept-encoding
```
Now the response through the hydra-router:
> http://localhost:5353/hapi-service-test/v1/welcome
```
Connection →keep-alive
Date →Wed, 19 Apr 2017 08:27:09 GMT
Transfer-Encoding →chunked
X-Hydra-Tracer →1lkglmzk4pn
```
As a result, the response from the router is unreadable because there are missing headers like content*

In my opinion, limiting the HTTP headers support on the hydra-router->hydra->UMF will limit the capabilities of the hydra ecosystem as well, at least on existing environments who want to get boosted by using hydra.

Here, I would suggest to introduce a "headers" field on the UMF spec, that could be used for communication headers like HTTP, in consequence it would make hydra-* more compatible on legacy environments.

Looking forward to your feedback.
Regards, Rolando.

**Comment by cjus** [comment]
@jkyberneees I'm doing some cleanup on Hydra-Router that will make this an easier conversation to have. Specifically, the HydraRouter `routeRequest` function is too large and in need of refactoring. Feedback on your points above to follow...

**Comment by jkyberneees** [comment]
Great, I keep an eye on this.

**Comment by cjus** [comment]
@jkyberneees please see if hydra-router 1.3.1 addresses this issue. Also requires that the service themselves using the latest hydra / hydra-express as updates are in makeAPIRequest.

**Comment by jkyberneees** [comment]
@cjus Hi Carlos, thanks for taking care of the issue/feature/suggestion. 
I reviewed the code and I see that you inject the request headers and consider the response too, however I don't understand this exclusion: https://github.com/flywheelsports/hydra/blob/master/index.js#L1103

Unfortunately, the feature is not yet working as expected, since hapi services are still not properly routed ;(
Let me know if you need more details to reproduce.

Best Regards, Rolando

**Comment by cjus** [comment]
@jkyberneees the line you referenced checks whether instanceList.length is zero - if that's the case then there are no more service instances to try so a service unavailable response is necessary.  The 1.3.1 release was found to have problems - please see if the issue is resolved in hydra-router:1.3.8
